src/Mod/Sketcher/Gui/TaskSketcherTool.h

```diff
@@ -0,0 +1,113 @@
+/***************************************************************************
+ *   Copyright (c) 2022 Pierre-Louis Boyer <pierrelouis.boyer@gmail.com>   *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef GUI_TASKVIEW_TaskSketcherTool_H
+#define GUI_TASKVIEW_TaskSketcherTool_H
+
+#include <Gui/TaskView/TaskView.h>
+#include <Gui/TaskView/TaskDialog.h>
+#include <Gui/Selection.h>
+#include <boost_signals2.hpp>
+#include <Inventor/events/SoKeyboardEvent.h>
+
+namespace App {
+class Property;
+}
+
+namespace Gui {
+class ViewProvider;
+}
+
+namespace SketcherGui {
+
+class Ui_TaskSketcherTool;
+class ViewProviderSketch;
+
+class SketcherToolWidget : public QWidget
+{
+    Q_OBJECT
+
+public:
+    SketcherToolWidget(QWidget *parent=0, ViewProviderSketch* sketchView=0);
+    ~SketcherToolWidget();
+
+    bool eventFilter(QObject* object, QEvent* event);
+
+    void loadSettings();
+
+    void setparameter(double val, int i);
+    void setLabel(const QString& atext, int i);
+    void setUnit(Base::Unit unit, int i);
+    void setParameterActive(bool val, int i);
+    void setParameterVisible(bool val, int i);
+    void setParameterFocus(int i);
+    void setSettings(int toolSelected);
+    bool isCheckBoxChecked(int i);
+    bool isWidgetActive;

```

Comment:
Now you can see if the widget is active from the DrawSketchHandler, just looking if the pointer to the tool widget is nullptr or not.